PhysicsTools/SelectorUtils/interface/VersionedSelector.h

```diff
@@ -150,42 +155,52 @@ initialize( const edm::ParameterSet& conf ) {
       << "You have supplied a null/empty cutflow to VersionedIDSelector,"
       << " please add content to the cuflow and try again.";
   }
+  
   // this lets us keep track of cuts without knowing what they are :D
   std::vector<edm::ParameterSet>::const_iterator cbegin(cutflow.begin()),
     cend(cutflow.end());
   std::vector<edm::ParameterSet>::const_iterator icut = cbegin;
-  for( ; icut != cend; ++icut ) {   
+  std::map<std::string,unsigned> cut_counter;
+  for( ; icut != cend; ++icut ) {  
+    std::stringstream realname;
     const std::string& name = icut->getParameter<std::string>("cutName");
+    if( !cut_counter.count(name) ) cut_counter[name] = 0;      
+    realname << name << "_" << cut_counter[name];
     const bool needsContent = 
       icut->getParameter<bool>("needsAdditionalProducts");     
     const bool ignored = icut->getParameter<bool>("isIgnored");
     candf::CandidateCut* plugin = 
-#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
       CutApplicatorFactory::get()->create(name,*icut);
-#else
-      nullptr;
-#endif
     if( plugin != nullptr ) {
-      cuts_.push_back(boost::shared_ptr<candf::CandidateCut>(plugin));
+      cuts_.push_back(std::shared_ptr<candf::CandidateCut>(plugin));
     } else {
       throw cms::Exception("BadPluginName")
 	<< "The requested cut: " << name << " is not available!";
     }
     needs_event_content_.push_back(needsContent);
-    this->push_back(name);
-    this->set(name);
-    if(ignored) this->ignoreCut(name);
+    const std::string therealname = realname.str();
+    this->push_back(therealname);
+    this->set(therealname);
+    if(ignored) this->ignoreCut(therealname);
+    cut_counter[name]++;
   }    
+
   //have to loop again to set cut indices after all are filled
   icut = cbegin;
+  cut_counter.clear();
   for( ; icut != cend; ++icut ) {
-    const std::string& name = icut->getParameter<std::string>("cutName");
-    cut_indices_.push_back(typename Selector<T>::index_type(&(this->bits_),name));
+    std::stringstream realname;
+    const std::string& name = icut->getParameter<std::string>("cutName");    
+    if( !cut_counter.count(name) ) cut_counter[name] = 0;      
+    realname << name << "_" << cut_counter[name];
+    cut_indices_.push_back(typename Selector<T>::index_type(&(this->bits_),realname.str()));    
+    cut_counter[name]++;
   }
+  
   initialized_ = true;
 }
 
-#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
+//#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
```

Comment:
Yes, it will break ROOT5 build.
I brought this up in the last ORP.
Now that there is actual code available in a PR form, we can discuss
this in ORP again

On 5/15/15 11:51 AM, Chris Jones wrote:
> In PhysicsTools/SelectorUtils/interface/VersionedSelector.h
> <https://github.com/cms-sw/cmssw/pull/9002#discussion_r30425224>:
> 
>>    initialized_ = true;
>>  }
>>  
>> -#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
>> +//#if !defined(__CINT__) && !defined(__MAKECINT__) && !defined(__REFLEX__)
> 
> Won't removing these break our ROOT 5 build?
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9002/files#r30425224>.
> 

-- 
***********************************************************************
 Vyacheslav (Slava) Krutelyov
 TAMU: Physics Dept Texas A&M MS4242, College Station, TX 77843-4242
 CERN: 42-R-027
 AIM/Skype: siava16              googleTalk: slava77@gmail.com
 (630) 291-5128  Cell (US)       +41 76 275 7116 Cell (CERN)

-----------------------------------------------------------------------